avocado/core/future/settings.py

```diff
@@ -0,0 +1,364 @@
+     configured by the user, on a more "permanent way".
+
+  3. Command-line options parsed in runtime. This is configured by the user, on
+     a more "temporary way";
+
+ATTENTION: This is a future module, and will be moved out from this package
+soon.
+"""
+
+import configparser
+import glob
+import os
+import re
+
+from pkg_resources import resource_filename
+
+from ..settings_dispatcher import SettingsDispatcher
+from ...utils import path
+
+
+class SettingsError(Exception):
+    """
+    Base settings error.
+    """
+
+
+class ConfigFileNotFound(SettingsError):
+    """
+    Error thrown when the main settings file could not be found.
+    """
+
+    def __init__(self, path_list):
+        super(ConfigFileNotFound, self).__init__()
+        self.path_list = path_list
+
+    def __str__(self):
+        return ("Could not find the avocado config file after looking in: %s" %
+                self.path_list)
+
+
+class AvocadoSettings:
+    """AvocadoSettings, an experimental Avocado configuration handler.
+
+    It is a simple wrapper around configparser and argparse.
+
+    Also, one object of this class could be passed as config to plugins and
+    modules.
+
+    Please, not that most of methods and attributes here are private. Only
+    public methods and attributes should be used outside this module.
+    """
+
+    def __init__(self, config_path=None):
+        """Constructor. Tries to find the main settings files and load them.
+
+        :param config_path: Path to a config file. Useful for unittesting.
+        """
+        self._config = configparser.ConfigParser()
+        self._all_config_paths = []
+        self._config_paths = []
+        self._short_mapping = {}
+        self._long_mapping = {}
+        self._namespaces = {}
+
+        # 1. Prepare config paths
+        if config_path is None:
+            self._prepare_base_dirs()
+            self._append_config_paths()
+        else:
+            # Only used by unittests (the --config parses the file later)
+            self._all_config_paths.append(config_path)
+
+        # 2. Parse/read all config paths
+        self._config_paths = self._config.read(self._all_config_paths)
+        if not self._config_paths:
+            raise ConfigFileNotFound(self._all_config_paths)
+
+    def _append_config_paths(self):
+        # 1. Append Defaults
+        self._append_pkg_defaults()
+
+        # 2. Override with system config
+        self._append_system_config()
+
+        # Allow plugins to modify/extend the list of configs
+        dispatcher = SettingsDispatcher()
+        if dispatcher.extensions:
+            dispatcher.map_method('adjust_settings_paths',
+                                  self._all_config_paths)
+
+        # 3. Override with the user's local config
+        self._append_user_config()
+
+    def _append_pkg_defaults(self):
+        config_pkg_base = os.path.join('etc', 'avocado', 'defaults.conf')
```

Comment:
@clebergnu `defaults.conf` is just a temporary file, during this process. `defaults.conf` will no be changed by users, for now. The idea is that, soon, we will have a `defaults.py` instead.
 
I agree with shipping a "commented configuration file", this will be the `avocado.conf` or `avocado.conf.example`. But this will be shipped also in the near future.

So, are you ok, with this `defaults.conf` for now?

Reply:
Yes, sure.  I understand this is an intermediary step, but it felt like a good idea to double (triple?) check the intention for the future.